src/State/Provider/ContentNegotiationProvider.php

```diff
@@ -67,7 +67,9 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
     private function addRequestFormats(Request $request, array $formats): void
     {
         foreach ($formats as $format => $mimeTypes) {
-            $request->setFormat($format, (array) $mimeTypes);
+            $existingMimeTypes = $request->getMimeTypes($format);
+            $newMimeTypes = array_unique(array_merge((array) $mimeTypes, $existingMimeTypes));
+            $request->setFormat($format, $newMimeTypes);
```

Comment:
> we think it'd be best to restore Symfony formats to the value it was initially when the request is done (for example in the ResponseProcessor ?). Doing this may allow some unwanted formats in the next request. ping @dunglas

Yeah I don't know the internals of API Platform well enough to say this fix won't have side-effects on how the system reads MIME types elsewhere, so I can understand the concern. To fix the best way possible I think will unfortunately be a fair bit more effort, because you'll need to read the MIME types for every supported format at the beginning of a request, store that state somewhere, then restore it at the end of every master request. Symfony doesn't provide an interface to iterate or read its internal formats mapping, you can only request it either per-format or per-mimetype on `Request`'s public API.

Reply:
definitely a good sign as we have some weird tests in there haha